infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/namespace/KubernetesNamespaceFactory.java

```diff
@@ -200,15 +121,37 @@ public boolean isManagingNamespace(String workspaceId) throws InfrastructureExce
    * @param workspaceId identifier of the workspace
    * @return created namespace
    */
-  public KubernetesNamespace create(String workspaceId, String namespace) {
-    return doCreateNamespace(workspaceId, namespace);
+  public KubernetesNamespace access(String workspaceId, String namespace) {
+    return doCreateNamespaceAccess(workspaceId, namespace);
   }
 
   @VisibleForTesting
-  KubernetesNamespace doCreateNamespace(String workspaceId, String name) {
+  KubernetesNamespace doCreateNamespaceAccess(String workspaceId, String name) {
     return new KubernetesNamespace(clientFactory, name, workspaceId);
   }
 
+  /**
+   * Checks if the current user is able to use the specified namespace for their new workspaces.
+   *
+   * @param namespaceName namespace name to check
+   * @throws ValidationException if the specified namespace is not permitted for the current user
+   */
+  public void checkIfNamespaceIsAllowed(String namespaceName) throws ValidationException {
+    if (allowUserDefinedNamespaces) {
+      // any namespace name is allowed but workspace start may fail
+      return;
+    }
+
+    String defaultNamespace =
+        evalPlaceholders(defaultNamespaceName, EnvironmentContext.getCurrent().getSubject(), null);
+    if (!namespaceName.equals(defaultNamespace)) {
+      throw new ValidationException(
+          "User defined namespaces are not allowed. You're able to specify only admin configured which is '"
```

Comment:
This message could be a little better-worded, something along the lines of 

```java
String.format("User defined namespaces are not allowed. Only the default namespace '%s' is available", defaultNamespaceName) 
```

maybe?